src/ocamlorg/lib/package.ml

```diff
@@ -58,58 +103,177 @@ module Info = struct
+    Lwt_fold.package_name_map
+      (fun name vmap acc ->
+        let+ vs =
+          Lwt_fold.package_version_map
+            (fun version opam acc ->
+              let package = OpamPackage.create name version in
+              let opam_file =
+                OpamPackage.Name.Map.find name opams
+                |> OpamPackage.Version.Map.find version
+              in
+              let+ rev_deps = mk_revdeps package packages rev_deps in
+              let t =
+                { synopsis =
+                    synopsis opam |> Option.value ~default:"No synopsis"
+                ; authors =
+                    author opam
+                    |> List.map (fun name ->
+                           Option.value
+                             (Opam_user.find_by_name name)
+                             ~default:(Opam_user.make ~name ()))
+                ; maintainers =
+                    maintainer opam
+                    |> List.map (fun name ->
+                           Option.value
+                             (Opam_user.find_by_name name)
+                             ~default:(Opam_user.make ~name ()))
+                ; license = license opam |> String.concat "; "
+                ; description =
+                    descr opam
+                    |> Option.map OpamFile.Descr.body
+                    |> Option.value ~default:""
+                ; homepage = homepage opam
+                ; tags = tags opam
+                ; rev_deps
+                ; conflicts = get_conflicts opam_file
+                ; dependencies = get_dependencies opam_file
+                ; depopts = get_depopts opam_file
+                ; url =
+                    url opam
+                    |> Option.map (fun url ->
+                           { uri = OpamUrl.to_string (OpamFile.URL.url url)
+                           ; checksum =
+                               OpamFile.URL.checksum url
+                               |> List.map OpamHash.to_string
+                           })
+                }
+              in
+              OpamPackage.Version.Map.add version t acc)
+            vmap
+            OpamPackage.Version.Map.empty
+        in
+        OpamPackage.Name.Map.add name vs acc)
+      opams
+      OpamPackage.Name.Map.empty
 end
 
 type t =
   { name : Name.t
   ; version : Version.t
-  ; info : Info.t lazy_t
+  ; info : Info.t
   }
 
 let name t = t.name
 
 let version t = t.version
 
-let info t = Lazy.force t.info
+let info t = t.info
 
-let create ~name ~version info = { name; version; info = Lazy.from_val info }
+let create ~name ~version info = { name; version; info }
 
 type state =
-  { mutable packages :
-      Info.t lazy_t OpamPackage.Version.Map.t OpamPackage.Name.Map.t
+  { version : string
+  ; mutable opam_repository_commit : string option
+  ; mutable packages : Info.t OpamPackage.Version.Map.t OpamPackage.Name.Map.t
   }
 
+let get_state { packages; _ } = packages
```

Comment:
The name of the function is a bit confusing. We have a type `state`, and the function `get_state` returns the packages of this state.

Reply:
Removed it as it's unused for now.